### ISSUE TITLE
`pnpm pack` is not documented

### ISSUE THREAD

**Issue by JasonKleban** [pnpm/pnpm#3084]
I made a package.json script called `pack` which was supposed to webpack something and I spent a fair amount of time troubleshooting why it was just doing a ton of work and no output and no webpack-like activity.  I believe it is an undocumented [pnpm command](https://github.com/pnpm/pnpm/blob/f9befbfb534e4027fd664b335eaf478ee6d890ea/packages/plugin-commands-publishing/src/pack.ts) intending to mimic `npm pack`?

1) This should be documented, right?
2) When running such a command that cannot be a script, should there be a banner or something?
3) Should pnpm warn when script conflict with builtins?
4) Is there an exhaustive list of reserved script names?

**Comment by wclr** [comment]
Btw noticed that pnpm pack  (probably publish) writes changed package.json and then reverts it back while performing its operations. Don't know if it is possible to get rid of these unsafe fs operations.

**Comment by zkochan** [comment]
why is it unsafe?
pnpm makes a lot of changes to package.json all the time. Why is it unsafe before publish? We need to convert the specs in package.json that uses workspace protocol to regular versions

**Comment by wclr** [comment]
> why is it unsafe? Why is it unsafe before publish? We need to convert the specs in package.json that uses workspace protocol to regular versions

I understands that there is need to placed a modified version in the published package. Because if something goes wrong during the process (it stops due to whatever and it stops unexpectedly) changed version maybe stay on fs being not reverted back, this may case a hassle. The second thing that there may be tools that react to changes of package files, not big deal but still unnecessary moves. But i'm not sure if it is possible to place in package files no from fs, didn't look at the code. Probably should be some way? But generally, the current way of doing this looks kinda dirty (not pure).

**Comment by zkochan** [comment]
Yes, it is a workaround because we use npm CLI under the hood to publish the package. Alternatively, we could copy the package to a temporary directory and publish it from there.

**Comment by wclr** [comment]
> Alternatively, we could copy the package to a temporary directory and publish it from there.

Ther is two paths to go:

1) Copy the content and run npm publish for that folder, but for packages with large content it may be not optimal. Also if you need to copy only package content files, there is [a package to get the list of them](https://github.com/npm/npm-packlist).

2) Seems that `npm publish` can take a ready-to-go tallbar, so it should be not difficult to prepare one again using `npm-packlist`.

**Comment by zkochan** [comment]
> Should pnpm warn when script conflict with builtins?

this might be a good idea

**Comment by zkochan** [comment]
This is implemented in current pnpm, so closing.

`pnpm pack` is documented at https://pnpm.io/cli/pack and `pnpm pack --help` describes it.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
